src/components/Dropdown.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState } from 'react'
+import { StyleSheet, Text, View } from 'react-native'
+import Touchable from 'src/components/Touchable'
+import DownArrowIcon from 'src/icons/DownArrowIcon'
+import UpArrowIcon from 'src/icons/UpArrowIcon'
+import Colors from 'src/styles/colors'
+import { typeScale } from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+
+interface Props<T> {
+  options: { value: T; label: string }[]
+  onValueSelected(value: T): void
+  testId?: string
+}
+
+/* Dropdown component with unscrollable list of options */
+
+function Dropdown<T>(props: Props<T>) {
+  const testID = props.testId ? props.testId : 'Dropdown'
+  const [isOpen, setIsOpen] = useState(false)
+  const [labelSelected, setLabelSelected] = useState<string | undefined>()
+
+  const toggleOpen = () => {
+    setIsOpen(!isOpen)
```

Comment:
```suggestion
    setIsOpen(prev => !prev)
```
this is safer in case of fast toggles / batched updates